lib/shopify_cli/theme/dev_server/app_extension_watcher.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+require "shopify_cli/file_system_listener"
+require "forwardable"
+
+module ShopifyCLI
+  module Theme
+    module DevServer
+      # Watches for file changes and publish events to the theme
+      class AppExtensionWatcher
+        extend Forwardable
+
+        def_delegators :@listener, :add_observer, :changed, :notify_observers
```

Comment:
I left these for usage in tests later, but currently only `add_observer` is being used.